firefox-ios/Client/Nimbus/NimbusSuggestFeatureLayer.swift

```diff
@@ -0,0 +1,33 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol NimbusSuggestFeatureLayerProtocol {
+    var config: [SuggestionType: Bool] { get }
+    func isAmpSuggestionAvailable() -> Bool
+    func isWikiSuggestionAvailable() -> Bool
+}
+
+/// A translation layer for the `firefoxSuggestFeature.fml`
+/// Responsible for creating a model for suggestion information available in the fml.
+class NimbusSuggestFeatureLayer: NimbusSuggestFeatureLayerProtocol {
+    let nimbus: FxNimbus
+
+    var config: [SuggestionType: Bool] {
+        nimbus.features.firefoxSuggestFeature.value().availableSuggestionsTypes
+    }
+
+    init(nimbus: FxNimbus = .shared) {
+        self.nimbus = nimbus
+    }
+
+    func isAmpSuggestionAvailable() -> Bool {
+        config[.amp] ?? false
+    }
+
+    func isWikiSuggestionAvailable() -> Bool {
+        config[.wikipedia] ?? false
+    }
```

Comment:
I have a suggestion for a slightly different way to write this, if you're open to it!

Adding separate `is*SuggestionAvailable()` methods for each provider definitely works, but also means that we need to add a method (and an `if` branch to `SearchViewController`) for every new provider we roll out.

I wonder if we can DRY out that approach a little bit.

What if we had a single method like this, that took a `SuggestionProvider` (you might need to `import Storage` at the top of the file to get `SuggestionProvider`):

```suggestion
    func isSuggestionProviderAvailable(_ provider: SuggestionProvider) -> Bool {
        return switch provider {
        case .amp: config[.amp] ?? false
        case .wikipedia: config[.wikipedia] ?? false
        default: false
        }
    }
```

...And then tweaked `SearchViewController` to use this as a filter predicate. I'll leave another comment there ☺️